editor/src/clj/editor/gui.clj

```diff
@@ -651,17 +649,18 @@
   (output node-outline-children [outline/OutlineData] :cached (g/fnk [child-outlines]
                                                                      (vec (sort-by :child-index child-outlines))))
   (output node-outline-reqs g/Any :cached (g/fnk []
-                                                 (mapv (fn [type-info] {:node-type (:node-cls type-info)
-                                                                        :tx-attach-fn (gen-gui-node-attach-fn (:node-type type-info))})
-                                                       (get-registered-node-type-infos))))
-  
+                                            (mapv (fn [type-info]
+                                                    {:node-type (:node-cls type-info)
+                                                     :tx-attach-fn (gen-gui-node-attach-fn (:node-type type-info))})
+                                                  (get-registered-node-type-infos))))
+
   (output node-outline outline/OutlineData :cached
           (g/fnk [_node-id id child-index node-outline-link node-outline-children node-outline-reqs type custom-type own-build-errors _overridden-properties]
             (cond-> {:node-id _node-id
                      :node-outline-key id
                      :label id
                      :child-index child-index
-                     :icon (or (:icon (get-registered-node-type-info type custom-type)) gui-icon)
+                     :icon (:icon (get-registered-node-type-info type custom-type))
```

Comment:
`get-registered-node-type-info` will now throw if the type isn't found. When loading, we catch this exception and mark the resource nodes as defective. Prior, the "unrecognized" data would be stripped from these resources when saving. Now, because they are defective, they will not be hooked up to the save data system, and trying to open one in the editor will produce an error dialog with a somewhat helpful error message ("Unable to open 'spine.collection', since it contains unrecognizable data. Could the project be missing a required extension?").